File: Unihub_app/lib/models/degree.dart
class Degree {
  final String name;
  final List<dynamic> subjects;

  Degree(
    this.name,
    this.subjects,
  );

  factory Degree.fromMap(Map<String, dynamic> json) {
    return Degree(
      json['name'],
      json['subjects'],
    );
  }
}

File: backend/src/models/Degree.ts
import {Schema,model,Document} from 'mongoose';

const schema = new Schema({
    name: String,
    subjects: Array,
}, {collection: 'degrees'});

interface IDegree extends Document {
    name: String;
    subjects: Array<String>;
}

export default model<IDegree>('Degree',schema);